src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Builder.cs

```diff
@@ -579,7 +579,9 @@ public void RemoveAt(int index)
             /// <param name="length">The number of elements to remove.</param>
             public void RemoveRange(int index, int length)
             {
-                Requires.Range(index >= 0 && index + length <= _count, nameof(index));
+                Requires.Range(index >= 0, nameof(index));
+                Requires.Range(length >= 0, nameof(length));
+                Requires.Range(index + length <= _count, nameof(index));
```

Comment:
```suggestion
                Requires.Range(index <= _count - length, nameof(index));
```
`index + length` could overflow otherwise.

There should also be a test for that case (overflow by e.g. int.MaxValue for one of them).